test/Microsoft.ML.Tests/TextLoaderTests.cs

```diff
@@ -840,6 +840,302 @@ public void TestTextLoaderBackCompat_VerWritt_0x0001000C()
+                labels[count] = labelPeriod;
+                features[count] = featuresPeriod.GetValues().ToArray();
+                count++;
+            }
+
+            // Read dataset with comma as decimal marker.
+            string dataPathDecimalMarkerComma = GetDataPath("iris-decimal-marker-as-comma.txt");
+            var readerDecimalMarkerComma = new TextLoader(mlContext, new TextLoader.Options()
+            {
+                Columns = new[]
+                        {
+                            new TextLoader.Column("Label", DataKind.UInt32, 0),
+                            new TextLoader.Column("Features", DataKind.Single, new [] { new TextLoader.Range(1, 4) }),
+                        },
+                DecimalMarker = ','
+            });
+            var textDataDecimalMarkerComma = readerDecimalMarkerComma.Load(GetDataPath(dataPathDecimalMarkerComma));
+
+            // Load values from iris-decimal-marker-as-comma.txt
+            DataViewSchema columnsComma = textDataDecimalMarkerComma.Schema;
+            using DataViewRowCursor cursorComma = textDataDecimalMarkerComma.GetRowCursor(columnsComma);
+            UInt32 labelComma = default;
+            ValueGetter<UInt32> labelDelegateComma = cursorComma.GetGetter<UInt32>(columnsComma[0]);
+            VBuffer<Single> featuresComma = default;
+            ValueGetter<VBuffer<Single>> featuresDelegateComma = cursorComma.GetGetter<VBuffer<Single>>(columnsComma[1]);
+
+            // Check values from iris-decimal-marker-as-comma.txt match those in iris.txt
+            count = 0;
+            while (cursorComma.MoveNext())
+            {
+                //Get values from respective columns
+                labelDelegateComma(ref labelComma);
+                featuresDelegateComma(ref featuresComma);
+                Assert.Equal(labels[count], labelComma);
+                Assert.Equal(features[count], featuresComma.GetValues().ToArray());
+                count++;
+            }
+        }
+
+        [Fact]
+        public void TestCommaAsDecimalMarkerDouble()
+        {
+            // Datasets iris.txt and iris-decimal-marker-as-comma.txt are the exact same, except for their
+            // decimal markers. Decimal marker in iris.txt is '.', and ',' in iris-decimal-marker-as-comma.txt.
+            // Here, the features are of type double, and the test checks for decimal markers with double.
+            var mlContext = new MLContext(seed: 1);
+
+            // Read dataset with period as decimal marker.
+            string dataPathDecimalMarkerPeriod = GetDataPath("iris.txt");
+            var readerDecimalMarkerPeriod = new TextLoader(mlContext, new TextLoader.Options()
+            {
+                Columns = new[]
+                        {
+                            new TextLoader.Column("Label", DataKind.UInt32, 0),
+                            new TextLoader.Column("Features", DataKind.Single, new [] { new TextLoader.Range(1, 4) }),
```

Comment:
I think you'd still need to change this to DataKind.Double for the getters to work correctly.